nemo_curator/utils/file_utils.py

```diff
@@ -45,7 +46,44 @@ def expand_outdir_and_mkdir(outdir):
     return outdir
 
 
-def get_all_files_paths_under(root, recurse_subdirectories=True, followlinks=False):
+def filter_files_by_extension(
+    files_list: List[str],
+    filter_by: Union[str, List[str]],
+):
+    """
+    Given a list of files, filter it to only include files matching given extension(s).
+
+    Args:
+        files_list: List of files.
+        filter_by: A string (e.g., "json") or a list of strings (e.g., ["json", "parquet"])
+            representing which file types to keep from files_list.
+
+    """
+    filtered_files = []
+
+    if isinstance(filter_by, str):
+        filter_by = [filter_by]
+
+    file_extensions = [s if s.startswith(".") else "." + s for s in filter_by]
+
+    for file in files_list:
+        if file.endswith(tuple(file_extensions)):
+            filtered_files.append(file)
+        else:
+            warning_flag = True
+
+    if warning_flag:
+        warnings.warn(f"Skipped at least one file due to unmatched file extension(s).")
```

Comment:
instead of tracking `warning_flag` as a boolean, you can just do `if len(files_list) != len(filtered_files): warnings.warn(...)`

Reply:
Great idea, thanks!